core/storage/audit/gae_models.py

```diff
@@ -79,3 +79,50 @@ def get_user_id_migration_policy():
     def get_user_id_migration_field(cls):
         """Return field that contains user ID."""
         return cls.user_id
+
+
+class UsernameChangeAuditModel(base_models.BaseModel):
```

Comment:
Assume I wanted to know all the usernames for a user with given user_id and current username, so how we will query on this model?

Also, assume I wanted to check the current username of a user whose old username was XYZ, so how the query will work?

[I would suggest creating a small doc that will compare alternative ways to store this info and their pros and cons.]

Reply:
Currently the model stores:
 - user_id: of the person(mostly admin) who makes the username change.
 - current_username
 - new_username

So, for your first question, I think the following is the way to do it:

```python
def get_usernames(user_id, current_username):
  query = cls.query(cls.user_id == user_id)
  return query.filter(cls.current_username == current_username)
```

And for the second question:

```python
def get_new_username_from_old(current_username):
  query = cls.query().filter(cls.current_username == current_username)
```

And then one can fetch the required info from the result of the query.

Does that answer your question?

Reply:
> Assume I wanted to know all the usernames for a user with given user_id and current username, so how we will query on this model?

Just to clarify, here the user_id and (current) username belongs to the same person. So you can't use `cls.query(cls.user_id == user_id)` as we tore the user_id of the person who has changed the username from the admin page, right?

Reply:
>  assume I wanted to check the current username of a user whose old username was XYZ, so how the query will work?
>> get_new_username_from_old(current_username)

Why this method takes current username as a param? (We are expecting that to be the output, right?)

Reply:
Also, wanted to put down the reason for these queries:
For case 1: Assume we wanted to check all the feedback provided by learners/reviewers for the given users' contributions. (Assuming they have added user's  username in the feedback)

For case 2: Assume we wanted to block a user with a given username and we didn't find any existing user with that username so we want to check the current_username of that user.

I'm not sure whether we want to have these features in this model (as it's going to be a small number of documents in this model). @seanlip, Wanted to check whether we want these features from this model? (If not, then I don't think it's worth spending time on this)

Reply:
We don't need these features in this model. This is just for auditing admin actions.

Reply:
Please note that, in general, nothing in the codebase should actually make use of this model, or of any models in this file.